include/aikido/planner/ConfigurationToEndEffectorOffsetPlanner.hpp

```diff
@@ -0,0 +1,42 @@
+#ifndef AIKIDO_PLANNER_CONFIGURATIONTOENDEFFECTOROFFSETPLANNER_HPP_
+#define AIKIDO_PLANNER_CONFIGURATIONTOENDEFFECTOROFFSETPLANNER_HPP_
+
+#include "aikido/planner/ConfigurationToEndEffectorOffset.hpp"
+#include "aikido/planner/SingleProblemPlanner.hpp"
+#include "aikido/trajectory/Trajectory.hpp"
+
+namespace aikido {
+namespace planner {
+
+/// Base planner class for ConfigurationToEndEffectorOffset planning problem.
+class ConfigurationToEndEffectorOffsetPlanner
+    : public SingleProblemPlanner<ConfigurationToEndEffectorOffsetPlanner,
+                                  ConfigurationToEndEffectorOffset>
+{
+public:
+  // Expose the implementation of Planner::plan(const Problem&, Result*) in
+  // SingleProblemPlanner. Note that plan() of the base class takes Problem
+  // while the virtual function defined in this class takes SolverbleProblem,
+  // which is simply ConfigurationToEndEffectorOffset.
+  using SingleProblemPlanner::plan;
+
+  /// Constructor
+  ///
+  /// \param[in] stateSpace State space that this planner associated with.
+  explicit ConfigurationToEndEffectorOffsetPlanner(
+      statespace::ConstStateSpacePtr stateSpace);
```

Comment:
This planner only works for `MetaSkeletonStateSpace`. Please change this class to:
* take `ConstMetaSkeletonStateSpace` in the constructor
* have a getter `getMetaSkeletonStateSpace()` that returns the `ConstStatespacePtr` by static-casting it to `ConstMetaSkeletonStateSpacePtr`.
to prevent passing incorrect state space in compile time.